### PR TITLE
feat: create a pytest plugin

### PR DESCRIPTION
**Pull request by guotin**
Created the plugin as in pytest docs

* Two cli flags for pytest, `--rts` and `--rts-init`. Maybe the `--rts-init` is pointless?
* Copied the logic from `select.py`
* `PYTEST_DONT_REWRITE` prevented the current implementation of collecting "newly added tests" from throwing an error
* Noticed an error of file being left open in `def line_mapping()` and tried to fix it
* Changed the tests to use the plugin
  * Would using `pytester` to test the plugin provide any benefits? (as in [pytest plugin docs](https://docs.pytest.org/en/stable/writing_plugins.html))
* I guess quite a bit of refactoring could be done now like removing the old scripts and useless files

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -27,7 +27,10 @@
             "pytest_rts_specific_without_remap=pytest_rts.tests.evaluation.specific_without_remap:main",
             "pytest_rts_all_without_remap=pytest_rts.tests.evaluation.all_without_remap:main",
             "pytest_rts_collect=pytest_rts.collect:main",
-        ]
+        ],
```
Comment (matveypashkovskiy) [inline setup.py:30]: shall we remove the lines above?

File: pytest_rts/plugin.py [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+        "--rts-init", action="store_true", default=False, help="run rts init"
+    )
+
+
+def pytest_configure(config):
+    """Register RTS plugins based on state"""
+    logger = logging.getLogger()
+    logging.basicConfig(format="%(message)s", level=logging.INFO)
+
+    if config.option.rts_init:
```
Comment (matveypashkovskiy) [inline pytest_rts/plugin.py:27]: shall we remove `--rts-init` option and just run init if there is no DB?

File: setup.py (hunk 1 above)
Comment (guotin) [reply setup.py:30]: At least the collection of newly added tests and the existing evaluation code (which I don't know if we're ever going to use anymore) relies on those.

File: pytest_rts/plugin.py (hunk 2 above)
Comment (guotin) [reply pytest_rts/plugin.py:27]: Possible if wanted. Have to also change the `conftest.py` to initialize the database correctly before every test.

File: setup.py (hunk 1 above)
Comment (matveypashkovskiy) [reply setup.py:30]: let's remove at least `pytest_rts` and `pytest_rts_init`

File: pytest_rts/plugin.py (hunk 2 above)
Comment (matveypashkovskiy) [reply pytest_rts/plugin.py]: in `conftest.py` replacing `subprocess.run(["pytest", "--rts-init"], check=True)` to something like:
* `subprocess.run(["rm", "coverage.db"], check=True)`
* `subprocess.run(["pytest", "--rts"], check=True)`

will do the job if `--rts` checks the existence of the DB

File: setup.py (hunk 1 above)
Comment (guotin) [reply setup.py:26]: Removed all the ones not related to new test collection or evaluation. Also removed the code that they were pointing to.

**Comment by matveypashkovskiy** [comment]
> Would using pytester to test the plugin provide any benefits? (as in pytest plugin docs)

Have no experience with it, maybe @alexey-vyskubov can answer. But in general, if you feel it is beneficial go ahead and use it.

> I guess quite a bit of refactoring could be done now like removing the old scripts and useless files

Would be nice

**Comment by matveypashkovskiy** [comment]
BTW @guotin could you please updated docs as well?

**Pull request merged**